server/app/src/main/app/bin/setenv.sh

```diff
@@ -1,20 +1,29 @@
-#!/bin/sh
-# ----------------------------------------------------------------------------
-# Copyright 2001-2010 The Apache Software Foundation.
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#      http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-# ----------------------------------------------------------------------------
-#
-# Add every needed extra jar to this
-CLASSPATH_PREFIX=../conf/lib/*
-export CLASSPATH_PREFIX
\ No newline at end of file
+#!/bin/sh
+# ----------------------------------------------------------------------------
+# Copyright 2001-2018 The Apache Software Foundation.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ----------------------------------------------------------------------------
+#
+# This file is sourced by the various start scripts. You can use it to
+# add extra environment variables to the startup procedure.
+#
+#   NOTE:  Instead of changing this file it is better to create a file
+#          named setenv.sh in the ../conf directory as the files in the
+#          bin directory should generally not be changed.
+
+# Add every needed extra jar to this
+CLASSPATH_PREFIX=../conf/lib/*
+export CLASSPATH_PREFIX
+
+[ -f "$BASEDIR"/conf/setenv.sh ] && . "$BASEDIR"/conf/setenv.sh
```

Comment:
I bet this file do not exist, right? While the location change looks good to me, I believe it is a bit confusing for users to have this file pointing to a non existing setenv.sh.

Maybe you can add it in `server/app/src/main/resources` ?

Reply:
Sure, but I currently do not understand how those files in the James distribution package, directory `conf`, get their `-template` name from?   Like for other config files, I do not want the product to ship with files in `conf` that may accidentally override customizations already made by the user.

Reply:
The **-template** suffix is shipped as part of the 3.0.1 but won't be in place for future releases (we intend to ship a software that starts without configuration changes.) So you don't need to worry about it.

If you add a file in that **conf** folder, it would only be available as part of new deployments and won't override anything for already deployed installation (you uncompress the tar only upon fist install).

Cheers,

Reply:
I AM not being childish